src/main/java/de/tum/in/www1/artemis/service/scheduled/UserScheduleService.java

```diff
@@ -0,0 +1,124 @@
+package de.tum.in.www1.artemis.service.scheduled;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.*;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.cache.CacheManager;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Service;
+
+import de.tum.in.www1.artemis.domain.User;
+import de.tum.in.www1.artemis.exception.VersionControlException;
+import de.tum.in.www1.artemis.repository.UserRepository;
+import de.tum.in.www1.artemis.service.connectors.VcsUserManagementService;
+
+@Service
+@Profile("scheduling")
+public class UserScheduleService {
+
+    @Value("${artemis.user-management.registration.cleanup-time-minutes}")
+    private Long removeNonActivatedUserDelayTime;
```

Comment:
Can this value be optional, so that Artemis will start without configuring this value?

Reply:
I could do that. I could also assign a default value of 60 minutes if it's not present?

Reply:
I like the default value idea.